Ports/cmake/patches/README.md

```diff
@@ -0,0 +1,244 @@
+- [ ] Should be merged to upstream? 
+- [X] Resolves issue(s) with our side of things
+- [X] Hack
+
+## `0018-isfinite.patch``
+
+Our `math.h` already defines `isfinite`.
+
+### Status
+- [ ] Local?
+- [X] Should be merged to upstream? If we want to have cmake support serenity out of the box.
+- [ ] Resolves issue(s) with our side of things
+- [ ] Hack
+
+## `0019-libuv-so_linger.patch`
+
+We don't have `SO_LINGER` or its associated struct. This patch removes them.
+
+### Status
+- [X] Local?
+- [ ] Should be merged to upstream?
+- [X] Resolves issue(s) with our side of things
+- [X] Hack
+
+## `0020-libuv-ipv6.patch`
+
+cmlibuv assumes the platform has definitions for (and supports) IPv6; this patch removes IPv6-specific things from libuv.
+
+### Status
+- [X] Local?
+- [ ] Should be merged to upstream?
+- [X] Resolves issue(s) with our side of things
+- [ ] Hack
+
+## `0021-libcurl-no-ipv6.patch`
+
+libcurl has a IPV6 switch, but uses `AF_INET6` unconditionally in one place. this patch disables IPV6 and makes that one use conditional.
+
+### Status
+- [X] Local? Partially.
+- [X] Should be merged to upstream? Partially.
+- [X] Resolves issue(s) with our side of things
+- [ ] Hack
+
+## `0022-remove-mutex.patch` and `0024-shared-mutex.patch`
+
+We don't have mutexes, and out libstdc++ does not define `std::mutex`.
+This patch removes all uses of `std::mutex`, `std::condition_variable`, and anything that depends on them; and replaces them with single-threaded equivalents.
+This will break horribly with smp.
+
+### Status
+- [X] Local?
+- [ ] Should be merged to upstream?
+- [X] Resolves issue(s) with our side of things
+- [X] Hack
+
+## `0023-lround.patch`
+
+Our libstdc++ (and stdlib) does not have `lround`, this patch replaces that with somewhat equivalent C stdlib functions.
+
+### Status
+- [X] Local?
+- [ ] Should be merged to upstream?
+- [X] Resolves issue(s) with our side of things
+- [X] Hack
+
+## `0025-uv-platform.patch`
+
+This patch adds the definitions necessary to compile libuv on Serenity.
+
+### Status
+- [ ] Local?
+- [X] Should be merged to upstream? If we want to have cmake support serenity out of the box.
+- [ ] Resolves issue(s) with our side of things
+- [ ] Hack
+
+## `0026-curl-struct-stat.patch`
+
+For unknown reasons, `curl_setup_once.h` does not include `sys/stat.h`. this patch includes `sys/stat.h`.
+
+### Status
+- [ ] Local?
+- [ ] Should be merged to upstream? 
+- [X] Resolves issue(s) with our side of things
+- [X] Hack
+
+## `0027-libuv-platform-serenity.patch`
+
+This patch adds a platform-specific conditional include to `uv/unix.h`.
+
+### Status
+- [ ] Local?
+- [X] Should be merged to upstream?
+- [ ] Resolves issue(s) with our side of things
+- [ ] Hack
+
+## `0028-cmake-disable-tests.patch`
+
+We don't care about building tests for now, and it makes the compilation much faster.
+
+### Status
+- [ ] Local?
+- [ ] Should be merged to upstream?
+- [ ] Resolves issue(s) with our side of things
+- [ ] Hack
```

Comment:
:100: for adding this file!

Reply:
It's a great idea to explain the patches, especially when they're non-trivial